components/script/task_queue.rs

```diff
@@ -186,8 +192,9 @@ impl<T: QueuedTaskConversion> TaskQueue<T> {
         self.msg_queue.borrow_mut().pop_front().ok_or(())
     }
 
-    /// Same as recv.
+    /// Retry to take tasks, and then recv.
     pub fn try_recv(&self) -> Result<T, ()> {
+        self.take_tasks(T::wake_up_msg());
         self.recv()
     }
```

Comment:
```suggestion
    /// Take all tasks again and then run `recv()`.
    pub(crate) fn take_tasks_and_recv(&self) -> Result<T, ()> {
        self.take_tasks(T::wake_up_msg());
        self.recv()
    }
```

Reply:
I know we had a conversation about this where I was a bit worried that this might allow starving the rest of the script thread, but as you say, you wrote this code and I trust you with this one.